itou/www/eligibility_views/forms.py

```diff
@@ -49,21 +49,21 @@ class AdministrativeCriteriaForm(forms.Form):
         f"Vous ne pouvez pas sélectionner en même temps les critères {NAME_DETLD_24} et {NAME_DELD_12}."
     )
 
-    administrative_criteria = AdministrativeCriteria.objects.all()
-
     def get_administrative_criteria(self):
-        return self.administrative_criteria
+        return AdministrativeCriteria.objects.all()
 
     def __init__(self, user, siae, **kwargs):
 
         self.user = user
         self.siae = siae
         super().__init__(**kwargs)
 
+        initial_administrative_criteria = self.initial.get("administrative_criteria", [])
         for criterion in self.get_administrative_criteria():
             key = criterion.key
             self.fields[key] = forms.BooleanField(required=False, label=criterion.name, help_text=criterion.desc)
             self.fields[key].widget.attrs["class"] = "form-check-input"  # Bootstrap CSS class.
+            self.initial.setdefault(key, criterion in initial_administrative_criteria)
```

Comment:
cryptic for me.
quel benefice cherches-tu dans cette modif ?

Reply:
Mon but étais de gérer le fait d'avoir des critères pré-remplis, mais je n'ai vu que (bien) plus tard comment c'étais fait ailleurs [1] donc j'ai tout de même garder cette méthode car je trouve que c'est plus naturel (ça été mon premier réflexe) de passer une liste `AdministrativeCriteria` :
```python
AdministrativeCriteriaForm(..., initial={"administrative_criteria": AdministrativeCriteria.object.all()})
```
Idéalement il faudrait changer l'autre bout de code pour n'avoir qu'une méthode mais de mémoire les objets ne sont pas les mêmes (`Evaluated...`) donc ça ne fonctionnais pas.


[1]
https://github.com/betagouv/itou/blob/c9aad699574082374b0eeda44675340756f733a8/itou/www/siae_evaluations_views/views.py#L271-L282